sql/2024/privacy/top_direct_ads_demand.sql

```diff
@@ -0,0 +1,63 @@
+WITH RECURSIVE pages AS (
+  SELECT
+    CASE page -- Publisher websites may redirect to an SSP domain, and need to use redirected domain instead of page domain. CASE needs to be replaced with a more robust solution from HTTPArchive/custom-metrics#136.
+      WHEN 'https://www.chunkbase.com/' THEN 'cafemedia.com'
+      ELSE NET.REG_DOMAIN(page)
+    END AS page,
+    JSON_QUERY(ANY_VALUE(custom_metrics), '$.ads') AS ads_metrics
+  FROM `httparchive.all.pages`
+  WHERE date = '2024-06-01' AND
+    is_root_page = TRUE
+  GROUP BY 1
+), ads AS (
+  SELECT
+    page,
+    JSON_QUERY(ads_metrics, '$.ads.account_types') AS ad_accounts
+  FROM pages
+  WHERE
+    JSON_VALUE(ads_metrics, '$.ads.account_count') != '0'
+), sellers AS (
+  SELECT
+    page,
+    JSON_QUERY(ads_metrics, '$.sellers.seller_types') AS ad_sellers
+  FROM pages
+  WHERE
+    JSON_VALUE(ads_metrics, '$.sellers.seller_count') != '0'
+), relationships AS (
+  SELECT
+    demand,
+    publisher,
+    COUNT(DISTINCT publisher) OVER () AS total_publishers
+  FROM (
+    SELECT
+      NET.REG_DOMAIN(domain) AS demand,
+      page AS publisher
+    FROM ads,
+      UNNEST(JSON_VALUE_ARRAY(ad_accounts, '$.direct.domains')) AS domain
+    UNION ALL
+    SELECT
+      page AS demand,
+      NET.REG_DOMAIN(domain) AS publisher
+    FROM sellers,
+      UNNEST(JSON_VALUE_ARRAY(ad_sellers, '$.publisher.domains')) AS domain
+    UNION ALL
+    SELECT
+      page AS demand,
+      NET.REG_DOMAIN(domain) AS publisher
+    FROM sellers,
+      UNNEST(JSON_VALUE_ARRAY(ad_sellers, '$.both.domains')) AS domain
+  )
+  GROUP BY
+    demand,
+    publisher
```

Comment:
SQLFluff says:
> L:  50 | P:   3 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY' clauses.

Anybody has an idea how to fix it?

Reply:
You have a `GROUP BY 1` on line 11 but then `GROUP BY`/`ORDER BY` column names later. Pick a style! :-)

Reply:
Oh, thanks. I thought it was related to that particular statement.

Reply:
Well technically it's all part of the one statement. It flagged it when it saw the second instance of `GROUP BY` and saw that it hadn't matched a previous instance of it.